### PR TITLE
Add pool share asset test up to the current protocol

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/test/TxTests.h [hunk 1]
```diff
@@ -255,6 +256,8 @@ Asset makeNativeAsset();
 Asset makeInvalidAsset();
 Asset makeAsset(SecretKey const& issuer, std::string const& code);
 Asset makeAssetAlphanum12(SecretKey const& issuer, std::string const& code);
+ChangeTrustAsset makePoolShareAsset(Asset const& assetA, Asset const& assetB,
```
Comment (jonjove) [inline src/test/TxTests.h:259]: Nitpick: not crazy about this name because it doesn't make it clear whether it is the hashed one or the raw one.

**Review by jonjove — COMMENTED** [review]
Mostly good, one minor comment. Thanks for adding this test.

File: src/test/TxTests.h (hunk 1 above)
Comment (sisuresh) [reply src/test/TxTests.h]: Updated to `makeChangeTrustAssetPoolShare`.

Comment (sisuresh) [reply src/test/TxTests.h]: I forgot to run `make format`...

Comment (sisuresh) [reply src/test/TxTests.h]: fixed

**Comment by jonjove** [comment]
r+ 869f0b9

**Pull request merged**